tests/test_localhost_integ.c

```diff
@@ -394,9 +394,13 @@ static int s_tester_on_put_body(struct aws_http_stream *stream, const struct aws
 
     (void)stream;
     (void)user_data;
-    struct aws_string *content_length_header_str = aws_string_new_from_cursor(s_tester.alloc, data);
-    s_tester.num_sen_received = (uint64_t)strtoull((const char *)content_length_header_str->bytes, NULL, 10);
-    aws_string_destroy(content_length_header_str);
+    /* Response is JSON: {"bytes": 2500000000} - extract the number */
+    const char *bytes_key = "\"bytes\": ";
+    const char *json_str = (const char *)data->ptr;
+    const char *bytes_pos = strstr(json_str, bytes_key);
+    if (bytes_pos != NULL) {
+        s_tester.num_sen_received = (uint64_t)strtoull(bytes_pos + strlen(bytes_key), NULL, 10);
+    }
```

Comment:
It's probably fine in test, but we are trying to not use `str` directly in the codebase, we have helpers in `byte_buf.h` and `string.h` in common.